arc4.c

```diff
@@ -173,13 +139,31 @@ arc4_ARC4_crypt(struct arc4_ARC4 *self, PyObject *arg)
     if (arg == NULL) {
         return NULL;
     }
-    if (byteslike_as_string_and_size(arg, (char **)&input, &size) == -1) {
-        if (PyErr_Occurred() == NULL) {
-            PyErr_Format(PyExc_TypeError,
-                         "crypt() argument 1 must be read-only bytes-like "
-                         "object, not %s",
-                         arg->ob_type->tp_name);
+    if (PyBytes_Check(arg)) {
+        input = PyBytes_AS_STRING(arg);
+        size = PyBytes_GET_SIZE(arg);
+    }
+    else if (PyUnicode_Check(arg)) {
+#if PY_MAJOR_VERSION <= 2
+        arg = PyUnicode_AsASCIIString(arg);
+        if (arg == NULL) {
+            return NULL;
         }
+        outputBytes = arc4_ARC4_crypt(self, arg);
+        Py_DECREF(arg);
+        return outputBytes;
```

Comment:
`PyUnicode_AsASCIIString()` converts `PyUnicodeObject` to `PyStringObject`, which passes `PyBytes_Check()` test, so this block doesn't recurse more than 1 time.